### ISSUE TITLE
Make file preview window read only

### ISSUE THREAD

**Issue by Thaodan** [alexluigit/dirvish#270]
### Thank you for the bug report

- [X] I am using the latest version of `dirvish` related packages.
- [X] I checked [CUSTOMIZING|EXTENSIONS](https://github.com/alexluigit/dirvish#resources).
- [X] You may also try reproduce the issue using clean environment and minimal configurations with
the command `emacs -Q`.

### Bug description

When opening Dirvish with file preview enabled the buffer containing the preview isn't read only.

### Steps to reproduce

1. Open dirvish
2. M-t
3. Preview image
4. Type inside image buffer

### Expected behavior

Buffer read-only

### OS

Linux

### Emacs Version

29

### Emacs Configurations

Vanilla

**Comment by alexluigit** [comment]
Why does the preview buffer has to be read-only?

**Comment by Thaodan** [comment]
Because you can just override the contents of the preview offer by
accident and if it read-only it also possible to bind keys without
prefixes such as q to close the preview.

It certainly not a requirement but it doesn't make sense coming from the
other Emacs modes which with special buffers such as Dired, image-dired
or any other mode which work more like an application inside Emacs
instead of those which operate on buffers backed by files.

**Comment by alexluigit** [comment]
Fair enough.  I implemented it in 8a2aafdc377453d8d5b92db6009a8ec4a131834d, for now we don't have any Dirvish specific keybindings attach to that preview window, maybe we can bind `q` to quit Dirvish (instead of quit the preview window only) and `e` to quit and edit the file being previewed, do you have any suggestions?

**Comment by Thaodan** [comment]
Isn't preview in a separate window? If so closing the preview and then
pressing q again to close dirvish should be fine or not? For the other
Emacs modes that's the way the closing mechanism works.

PS: Since you closed this issue, should we open another one for the
keybindings in the preview frame?
